app/react/Users/components/specs/UserSidePanel.spec.tsx

```diff
@@ -0,0 +1,210 @@
+      { field: 'email', value: '', message: 'Email is required' },
+      { field: 'password', value: 'a'.repeat(55), message: 'Password is too long' },
+    ])(
+      'should not save if there is an invalid value %s',
+      ({ field, value, message }, done: jest.DoneCallback) => {
+        const props = { ...defaultProps };
+        props.user = { ...newUser, [field]: value };
+        const wrapper = render(props);
+        wrapper.find('form').simulate('submit');
+
+        setImmediate(() => {
+          wrapper.update();
+          const error = wrapper
+            .find({ id: `${field}_field` })
+            .children()
+            .find('div')
+            .at(0);
+          expect(defaultProps.onSave).not.toBeCalled();
+          expect(error.text()).toEqual(message);
+          done();
+        });
+      }
+    );
+
+    it('should save the changes over the user', done => {
+      const emailInput = component.find({ id: 'email_field' }).find('input');
+      const fakeEvent = {
+        currentTarget: { name: 'email', value: 'newemail@test.test' },
+      };
+      // @ts-ignore
+      emailInput.prop('onChange')!(fakeEvent);
+      component.find('form').simulate('submit');
+      const savedUser = { ...existingUser, email: 'newemail@test.test' };
+      setImmediate(() => {
+        expect(defaultProps.onSave).toHaveBeenCalledWith(savedUser);
+        done();
+      });
+    });
+  });
+
+  describe('Reset 2FA', () => {
+    it('should not show reset 2FA action button if user is not using 2fa', () => {
+      expect(component.find('#reset2faBtn').length).toBe(0);
+    });
+    describe('User using 2fa', () => {
+      it('should call the onReset2fa function when confirm reset 2fa', () => {
+        const props = { ...defaultProps };
+        props.user.using2fa = true;
+        component = render(props);
+        component.find('#reset2faBtn').simulate('click');
+        component.find('.confirm-button').simulate('click');
+        expect(defaultProps.onReset2fa).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('Reset Password', () => {
+    it('should call the resetPasswordHandler function when confirm reset password', () => {
+      component.find('#resetPasswordBtn').simulate('click');
+      component.find('.confirm-button').simulate('click');
+      expect(defaultProps.onResetPassword).toHaveBeenCalled();
+    });
+  });
+
+  describe('Groups membership', () => {
+    it('should list all the available groups sorted alphabetically', () => {
+      const availableGroups = component.find(MultiSelect);
+      expect(availableGroups.props().options).toEqual([group1, group2]);
+    });
+
+    describe('adding a group', () => {
+      beforeEach(() => {
+        const groupsToCheck = component
+          .find(MultiSelect)
+          .find('input')
+          .at(1);
+        groupsToCheck.simulate('change');
+      });
+
+      it('should add a checked user to the selected users', () => {
```

Comment:
This description is not clear, can we re-word it?

Reply:
Right, done